commands/board/list.go

```diff
@@ -50,6 +51,39 @@ var (
 	validVidPid = regexp.MustCompile(`0[xX][a-fA-F\d]{4}`)
 )
 
+func cachedApiByVidPid(vid, pid string) ([]*rpc.BoardListItem, error) {
+	var resp []*rpc.BoardListItem
+
+	cacheKey := fmt.Sprintf("cache.builder-api.v3/boards/byvid/pid/%s/%s", vid, pid)
+	if cachedResp := inventory.Store.GetString(cacheKey + ".data"); cachedResp != "" {
+		ts := inventory.Store.GetTime(cacheKey + ".ts")
+		if time.Since(ts) < time.Hour*24 {
+			// Use cached response
+			if cachedResp == "ErrNotFound" {
+				return nil, ErrNotFound
```

Comment:
is it correct to return this?

Reply:
Yes, the web API is ill-defined, it returns a 404 (HTTP Not Found) if the vid/pid is not in the database.
It would have been better a 200 (HTTP OK) with an empty response, but I think it's late to change it now.

Reply:
or maybe we can mask this behavior inside `apiByVidPid`? this looks like a good idea, can make the whole thing a bit nicer, let me try it, I'll add a new commit to this PR